src/rgw/rgw_rados.cc

```diff
@@ -8354,8 +8354,12 @@ int RGWRados::delete_system_obj(rgw_obj& obj, RGWObjVersionTracker *objv_tracker
   return 0;
 }
 
-int RGWRados::delete_obj_index(rgw_obj& obj)
+int RGWRados::delete_obj_index(rgw_obj& _obj)
 {
+  rgw_obj obj = _obj;
+  if (obj.get_instance() == "null") {
```

Comment:
@zhangsw I'm not sure this is correct. There is a difference between deleting an object and having the null version specified, and deleting the object when no version was specified. In a versioned bucket, if a null version has existed before (e.g., from before the bucket versioning was set), then having 'null' as version will remove that object's instance, whereas not having it will create a new delete marker.